File: src/main/java/org/hamster/service/ReportDiff.java
package org.hamster.service;

import liquibase.diff.DiffResult;
import liquibase.diff.output.report.DiffToReport;
import liquibase.exception.DatabaseException;
import org.hamster.enumeration.Extension;
import org.hamster.config.DbConfiguration;

import java.io.PrintStream;

public class ReportDiff extends AbstractDiff {

    public ReportDiff(DbConfiguration dbConfiguration) {
        super(dbConfiguration);
    }

    @Override
    public void diff(DiffResult diffResult, PrintStream out) throws DatabaseException {
        DiffToReport diffToReport = new DiffToReport(diffResult, out);
        diffToReport.print();
    }

    @Override
    public String getExtension() {
        return Extension.TXT.getExtension();
    }
}

File: src/main/kotlin/org/hamster/service/ReportDiff.kt
package org.hamster.service

import liquibase.diff.DiffResult
import liquibase.diff.output.report.DiffToReport
import org.hamster.config.DbConfiguration
import org.hamster.enumeration.Extension
import java.io.PrintStream

class ReportDiff(dbConfiguration: DbConfiguration?) : AbstractDiff(dbConfiguration!!) {
    override fun diff(diffResult: DiffResult, out: PrintStream) {
        val diffToReport = DiffToReport(diffResult, out)
        diffToReport.print()
    }

    override val extension: String
        get() = Extension.TXT.extension
}